plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,538 @@
+        description=dict(type="str"),
+        version3_asm=dict(type="str", aliases=["allow_version3_asm"], choices=["enabled", "disabled"]),
+        fast_leave=dict(type="str", choices=["enabled", "disabled"]),
+        report_link_local_groups=dict(type="str", choices=["enabled", "disabled"]),
+        igmp_version=dict(type="str", choices=["v2", "v3"]),
+        group_timeout=dict(type="int"),
+        query_interval=dict(type="int"),
+        query_response_interval=dict(type="int"),
+        last_member_count=dict(type="int"),
+        last_member_response_time=dict(type="int"),
+        startup_query_count=dict(type="int"),
+        startup_query_interval=dict(type="int"),
+        querier_timeout=dict(type="int"),
+        robustness_variable=dict(type="int"),
+        state_limit_route_map_uuid=dict(type="str", aliases=["state_limit_route_map_for_multicast"]),
+        state_limit_route_map=dict(
+            type="dict",
+            options=dict(
+                name=dict(type="str"),
+            ),
+            aliases=["state_limit_route_map_policy", "state_limit_route_map_policy_multicast"],
+        ),
+        report_policy_route_map_uuid=dict(type="str", aliases=["report_policy_route_map_for_multicast"]),
+        report_policy_route_map=dict(
+            type="dict",
+            options=dict(
+                name=dict(type="str"),
+            ),
+            aliases=["report_policy_route_map_policy", "report_policy_route_map_policy_multicast"],
+        ),
+        static_report_route_map_uuid=dict(type="str", aliases=["static_report_route_map_for_multicast"]),
+        static_report_route_map=dict(
+            type="dict",
+            options=dict(
+                name=dict(type="str"),
+            ),
+            aliases=["static_report_route_map_policy", "static_report_route_map_policy_multicast"],
+        ),
+        maximum_multicast_entries=dict(type="int"),
+        reserved_multicast_entries=dict(type="int"),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["template", "template_id"], True],
+            ["state", "query", ["template", "template_id"], True],
+            ["state", "absent", ["template", "template_id"], True],
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+        mutually_exclusive=[
+            ["template", "template_id"],
+            ["state_limit_route_map_uuid", "state_limit_route_map"],
+            ["report_policy_route_map_uuid", "report_policy_route_map"],
+            ["static_report_route_map_uuid", "static_report_route_map"],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template_identifier = get_object_identifier(module.params.get("template_id"), module.params.get("template"))
```

Comment:
Can you refer https://github.com/CiscoDevNet/ansible-mso/pull/615/files and make changes here?

Reply:
fixed